File: aeds2/test/Main.java
import java.util.Scanner;

public class Main {
    public static boolean palindrome(String value) {
        int valueSize = value.length() - 1;
        boolean resp = false;
        String symbols = "!@#$%¨&*-'`~^";

        for (int i = 1; i <= valueSize; i++) {
            for (int j = 1; j <= symbols.length() -1 ; j++) {
                if (value.charAt(i) == symbols.charAt(j)) {
                    System.out.println(value.charAt(i));
                }
            }

            if (value.charAt(i) != value.charAt(valueSize - i)) {
                resp = false;
            } else {
                resp = true;
            }
        }
        return resp;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String value;

        while (sc.hasNext()) {
            value = sc.nextLine();

            if (palindrome(value)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
        }

        sc.close();
    }
}

File: aeds2/test/main.c
#include <stdio.h>
#include <string.h>

int palindrome(char string[]);

int main(void) 
{
    char inputFile[100];

    while (scanf("%s", inputFile) != EOF)
    {
        int isPalindrome = palindrome(inputFile);

        if (isPalindrome == 0) {
            printf("NÃO | %s\n", inputFile);
        } else
        {
            printf("SIM | %s\n", inputFile);
        }
    }
}

int palindrome(char string[])
{
    int length = strlen(string);
    int resp = 0; // 0 for false and 1 for true

    for (int i = 0; i < length - 1; i++)
    {
        if (string[i] != string[length - i])
        {
            resp = 0;
        } else 
        {
            resp = 1;
        }
    }
    return resp;
}
